ilc/client/ClientRouter.js

```diff
@@ -212,29 +212,34 @@ export default class ClientRouter extends EventEmitter {
         const appsWithDifferentProps = this.#getAppsWithDifferentProps(this.#prevRoute.slots, this.#currentRoute.slots);
         if (appsWithDifferentProps.length) {
             const appsToForceRerender = [];
+            const updateEventsToTrigger = [];
 
             appsWithDifferentProps.forEach(({ slotName, appName }) => {
                 const eventUpdateFragment = `ilc:update:${slotName}_${appName}`;
 
                 // if fragment provided "update" lifecycle method then it will be updated immediately w/o remounting app
                 // otherwise the fragment will be unmounted and mounted with new props
                 if (this.listenerCount(eventUpdateFragment)) {
-                    this.emit(eventUpdateFragment);
+                    updateEventsToTrigger.push(eventUpdateFragment);
                 } else {
                     // temporary remove slot with old props, to remove it from DOM
                     // it will be rendered with new props with the help of "triggerAppChange"
-                    appsToForceRerender.push({ slotName, appName })
+                    appsToForceRerender.push(appName)
                     delete this.#activeApps.current[slotName];
                 }
             });
 
-            if (appsToForceRerender.length) {
-                window.addEventListener('single-spa:app-change', () => {
-                    this.#logger.log(`ILC: Triggering app re-mount for [${appsToForceRerender.map(n => n.appName)}] due to changed props.`);
+            window.addEventListener('single-spa:app-change', () => {
+                if (updateEventsToTrigger.length) {
+                    updateEventsToTrigger.forEach(this.emit.bind(this));
```

Comment:
as far as I understand `triggerAppChange` emits `ilc:before:routing`, probably emmiting update events before `ilc:before:routing` is triggered wouldn't be correct

Reply:
done